yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -513,13 +517,16 @@ private[spark] class ApplicationMaster(
         logInfo("Waiting for spark context initialization ... ")
         sparkContextRef.wait(10000L)
       }
-
-      val sparkContext = sparkContextRef.get()
-      if (sparkContext == null) {
-        logError(("SparkContext did not initialize after waiting for %d ms. Please check earlier"
-          + " log output for errors. Failing the application.").format(totalWaitTime))
+      if (!finished) {
+        val sparkContext = sparkContextRef.get()
+        if (sparkContext == null) {
+          logError(("SparkContext did not initialize after waiting for %d ms. Please check earlier"
+            + " log output for errors. Failing the application.").format(totalWaitTime))
+        } else {
+          Some(sparkContext)
+        }
       }
-      sparkContext
+      None
```

Comment:
hi @jerryshao 
I am not so good in Scala coding, infact this is the first or second time I am touching scala code :)
I expected the previous line Some(sparkContext) is going to do the trick but, looks like I need to fix that.